vector/src/main/java/im/vector/app/features/settings/devices/v2/ParseDeviceUserAgentUseCase.kt

```diff
@@ -0,0 +1,147 @@
+            userAgent.contains(IOS_KEYWORD) -> parseIosUserAgent(userAgent)
+            userAgent.contains(DESKTOP_KEYWORD) -> parseDesktopUserAgent(userAgent)
+            userAgent.contains(WEB_KEYWORD) -> parseWebUserAgent(userAgent)
+            else -> createUnknownUserAgent()
+        }
+    }
+
+    private fun parseAndroidUserAgent(userAgent: String): DeviceExtendedInfo {
+        val appName = userAgent.substringBefore("/")
+        val appVersion = userAgent.substringAfter("/").substringBefore(" (")
+        val deviceInfoSegments = userAgent.substringAfter("(").substringBeforeLast(")").split("; ")
+        val deviceModel: String?
+        val deviceOperatingSystem: String?
+        if (deviceInfoSegments.firstOrNull() == "Linux") {
+            val deviceOperatingSystemIndex = deviceInfoSegments.indexOfFirst { it.startsWith("Android") }
+            deviceOperatingSystem = deviceInfoSegments.getOrNull(deviceOperatingSystemIndex)
+            deviceModel = deviceInfoSegments.getOrNull(deviceOperatingSystemIndex + 1)
+        } else {
+            deviceModel = deviceInfoSegments.getOrNull(0)
+            deviceOperatingSystem = deviceInfoSegments.getOrNull(1)
+        }
+        return DeviceExtendedInfo(DeviceType.MOBILE, deviceModel, deviceOperatingSystem, appName, appVersion)
+    }
+
+    private fun parseIosUserAgent(userAgent: String): DeviceExtendedInfo {
+        val appName = userAgent.substringBefore("/")
+        val appVersion = userAgent.substringAfter("/").substringBefore(" (")
+        val deviceInfoSegments = userAgent.substringAfter("(").substringBeforeLast(")").split("; ")
+        val deviceModel = deviceInfoSegments.getOrNull(0)
+        val deviceOperatingSystem = deviceInfoSegments.getOrNull(1)
+        return DeviceExtendedInfo(DeviceType.MOBILE, deviceModel, deviceOperatingSystem, appName, appVersion)
+    }
+
+    private fun parseDesktopUserAgent(userAgent: String): DeviceExtendedInfo {
+        val browserSegments = userAgent.split(" ")
+        val browserName = when {
+            isFirefox(browserSegments) -> {
+                "Firefox"
+            }
+            isEdge(browserSegments) -> {
+                "Edge"
+            }
+            isMobile(browserSegments) -> {
+                getMobileBrowserName(browserSegments)
+            }
+            isSafari(browserSegments) -> {
+                "Safari"
+            }
+            else -> {
+                getRegularBrowserName(browserSegments)
+            }
+        }
+
+        val deviceOperatingSystemSegments = userAgent.substringAfter("(").substringBefore(")").split("; ")
+        val deviceOperatingSystem = if (deviceOperatingSystemSegments.getOrNull(1)?.startsWith("Android").orFalse()) {
+            deviceOperatingSystemSegments.getOrNull(1)
+        } else {
+            deviceOperatingSystemSegments.getOrNull(0)
+        }
+        return DeviceExtendedInfo(DeviceType.DESKTOP, browserName, deviceOperatingSystem, null, null)
```

Comment:
I think in case of the Web device type, we should set browser info into the `clientName` and `clientVersion`. It seems we should only use these fields for the case of Web devices to render a "Browser" section. For the section "Application", we will use info coming from `accountData` event.